src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -557,6 +557,9 @@ internal object ThrownException
             internal volatile UIntPtr _handlingFrameSP;
 
 #if TARGET_ARM64
+            // On ARM64, two frames can have the same SP, when a leaf function
```

Comment:
Does this actually happen in our current codegen? I thought that we always emit non-empty frames currently:

```
            stp     fp, lr, [sp, #-0x10]!
            mov     fp, sp
            ldp     fp, lr, [sp], #0x10
            ret     lr
```

If this is a future proofing in case we stop doing this, I think this is needed on all current architectures except x86/x64.

Reply:
This was there since May 2018. This PR doesn't change it in any way. Here is the PR that brought that in: https://github.com/dotnet/corert/pull/5560. So it is possible that it is not needed for NativeAOT / CoreCLR and it is just a remainder of handling the behavior of the pre-nativeaot compiler.
Let me run coreclr / libraries tests on arm64 device with this removed to see if there are any issues.